File: Components/GildedRose.Logic/Models/Inventory.cs
using System;

namespace GildedRose.Logic.Models
{
    public class Inventory
    {
        public Guid Identifier { get; set; }

        public string Name { get; set; }

        public int CategoryId { get; set; }

        public string CategoryName { get; set; }

        public int InitialQuality { get; set; }

        public int CurrentQuality { get; set; }

        public int MaxQuality { get; set; }

        public int SellIn { get; set; }

        public bool IsLegendary { get; set; }

        public DateTime Created { get; set; }

        public int CreatedBy { get; set; }

        public DateTime? Modified { get; set; }

        public int? ModifiedBy { get; set; }
    }
}

File: Web/GildedRose.Client/GildedRose.App/src/app/models/Inventory.ts
export interface InventoryModel {
  identifier: string;
  name: string;
  categoryId: number;
  categoryName: string;
  initialQuality: number;
  currentQuality: number;
  maxQuality: number;
  sellIn: number;
  isLegendary: boolean;
}

export namespace InventoryModel {
  export enum Filter {
    SHOW_ALL = "all",
    SHOW_ACTIVE = "trash",
  }
}
